CRFModuleValidation/GroupedScheduledActivityManager.swift

```diff
@@ -0,0 +1,149 @@
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+import BridgeSDK
+import BridgeAppSDK
+import ResearchUXFactory
+
+class GroupedScheduledActivityManager : ScheduledActivityManager {
+    
+    // Set a default to the all activities
+    var taskGroup: TaskGroup! = TaskGroup.addActivities
+    var timingActivity: SBBScheduledActivity?
+    var timingActivityFired: Bool = false
+    
+    // Set the date for activities to show. default == today.
+    var date: Date = Date()
+    
+    var allActivitiesCompleted: Bool {
+        return self.activities.reduce(true, { $0 && $1.finishedOn != nil })
+    }
+    
+    var shouldFireTimingSchedule: Bool {
+        return false
+//        return !timingActivityFired &&
+//            (timingActivity != nil) &&
+//            allActivitiesCompleted &&
+//            MasterScheduledActivityManager.shared.shouldFireTimingSchedule(for: self.taskGroup)
+    }
+    
+    func createTimingTaskViewController() -> ORKTaskViewController? {
+        guard let schedule = timingActivity else { return nil }
+        let taskVC = self.createTaskViewController(for: schedule)
+        timingActivityFired = true
+        return taskVC
+    }
+    
+    override var activities:[SBBScheduledActivity] {
+        get { return super.activities }
+        set {
+            super.activities = taskGroup.filteredAndSorted(newValue, on: date)
+//            self.timingActivity = taskGroup.timingSchedule(from: newValue)
+        }
+    }
+    
+//    override func shouldIncludeTimingIntroduction(for timingSchedule: SBBScheduledActivity) -> Bool {
+//        guard let lhs = timingSchedule.taskId, let rhs = self.timingActivity?.taskId else { return false }
+//        return lhs == rhs && allActivitiesCompleted
+//    }
+    
+    override func reloadData() {
```

Comment:
Note: this will block any activity from reloading that changes due to this task changing. Basically, this was b/c we already had all the sub schedule activities and didn't need to go get them again.

Reply:
Good to know.